tests/flow_control.rs

```diff
@@ -380,55 +380,31 @@ fn stream_close_by_data_frame_releases_capacity() {
         h2.unwrap()
     });
 
-    let mock = mock.assert_client_handshake().unwrap()
-        // Get the first frame
-        .and_then(|(_, mock)| mock.into_future().unwrap())
-        .and_then(|(frame, mock)| {
-            let request = assert_headers!(frame.unwrap());
-
-            assert_eq!(request.stream_id(), 1);
-            assert!(!request.is_end_stream());
-
-            mock.into_future().unwrap()
-        })
-        .and_then(|(frame, mock)| {
-            let request = assert_headers!(frame.unwrap());
-
-            assert_eq!(request.stream_id(), 3);
-            assert!(!request.is_end_stream());
-
-            mock.into_future().unwrap()
-        })
-        .and_then(|(frame, mock)| {
-            let data = assert_data!(frame.unwrap());
-
-            assert_eq!(data.stream_id(), 1);
-            assert_eq!(data.payload().len(), 0);
-            assert!(data.is_end_stream());
-
-            mock.into_future().unwrap()
-        })
-        .and_then(|(frame, _)| {
-            let data = assert_data!(frame.unwrap());
-
-            assert_eq!(data.stream_id(), 3);
-            assert_eq!(data.payload(), "hello");
-            assert!(data.is_end_stream());
-
-            Ok(())
-        });
-
-    let _ = h2.join(mock).wait().unwrap();
+    let srv = srv.assert_client_handshake().unwrap()
+        .ignore_settings()
+        .recv_frame(
+            frames::headers(1).request("POST", "https://http2.akamai.com/")
+        )
+        .send_frame(frames::headers(1).response(200))
```

Comment:
The code being replaced didn't send frames back, is this difference important?

Reply:
It's not currently, but it will be – in order to make these tests work with the reset-on-drop PR, the client will need to actually read frames back (to prevent the streams from being dropped).